vhdbuilder/packer/pre-install-dependencies.sh

```diff
@@ -48,11 +48,47 @@ rm -f /etc/cron.daily/logrotate
 
 systemctlEnableAndStart sync-container-logs.service || exit 1
 
-if [[ (${UBUNTU_RELEASE} == "20.04" || ${UBUNTU_RELEASE} == "18.04" || ($OS == $MARINER_OS_NAME && $OS_VERSION == "2.0")) && ${ENABLE_FIPS,,} == "true" ]]; then
-  installFIPS
-elif [[ ${ENABLE_FIPS,,} == "true" ]]; then
-  echo "AKS enables FIPS on Ubuntu 18.04, 20.04 or Mariner 2.0 only, exiting..."
-  exit 1
+# First handle Mariner + FIPS
+if [[ ${OS} == ${MARINER_OS_NAME} ]]; then
+  if [[ ${ENABLE_FIPS,,} == "true" ]]; then
+    # This is FIPS install for Mariner and has nothing to do with Ubuntu Advantage
+    echo "Install FIPS for Mariner SKU"
+    installFIPS
+  fi
+else
+  # Handle FIPS and ESM for Ubuntu
+  useUA=$( [[ "${UBUNTU_RELEASE}" = "18.04" ]] || [[ "${ENABLE_FIPS,,}" = "true" ]] && echo "true" || echo "false" )
```

Comment:
nit: I think this is actually more chars than just doing it inline :D

Reply:
wait also what is this echo...

Reply:
oops I was testing the one liner

Reply:
I was gonna use useUA twice, one to autoattach and once to detach, hence thought of a one liner, but yeah, inline works too, let me update.

Reply:
resolved.